### ISSUE TITLE
Problems while creating thumbnails in production

### ISSUE THREAD

**Issue by mcrot** [ContactEngineering/TopoBank#782]
See today when recreating all images:
```
2022-01-17T13:25:57.666584767Z prodstack_celeryworker.1.m78k38bvikrz@analyses    | [2022-01-17 14:25:57,665: WARNING/ForkPoolWorker-33] Problems while generating thumbnail for topography 570: Message: Failed to decode response from marionette
[...]
2022-01-17T13:25:57.779540899Z prodstack_celeryworker.1.m78k38bvikrz@analyses    | WARNING 2022-01-17 14:25:57,779 models 117 140065250571648 Problems while generating thumbnail for topography 416: Message: TypeError: this.curBrowser.messageManager is null
2022-01-17T13:25:57.779593488Z prodstack_celeryworker.1.m78k38bvikrz@analyses    | . Saving <None> instead.
```
One is for topography 570, the other for topography 416.

**Comment by mcrot** [comment]
This happens for many topographies. Probably related to https://github.com/mozilla/geckodriver/issues/1812.
The firefox  + geckodriver currently installed are
```
firefox                   79.0
geckodriver               0.26.0
```

**Comment by mcrot** [comment]
Another message which happens here for an 8kx8k topography (id: 819):
```
2022-01-21T13:43:10.683410406Z prodstack_celeryworker.1.0ms4qtvswwj0@analyses    | WARNING 2022-01-21 14:43:10,682 models 124 139985234109824 Problems while generating thumbnail for topography 819: Message: invalid argument: can't kill an exited process
2022-01-21T13:43:10.683468560Z prodstack_celeryworker.1.0ms4qtvswwj0@analyses    | . Saving <None> instead.
```
With 33GB available on /tmp, almost no tasks running, at least 40GB of free RAM.
As image the topography needs roughly 250 MB, so probably memory is not the problem here.

It seems to already happens during thumbnail generation, because the thumbnail isn't there. The "Saving <None>" message is also about the thumbnail.

**Comment by mcrot** [comment]
I've put in a traceback in order to see what's happening:
```python
2022-01-21T14:24:24.618586366Z prodstack_celeryworker.1.lki36nzyw4k2@analyses    | [2022-01-21 15:24:24,617: WARNING/ForkPoolWorker-32] Problems while generating thumbnail for topography 819: Message: Timeout loading page after 300000ms
2022-01-21T14:24:24.618591106Z prodstack_celeryworker.1.lki36nzyw4k2@analyses    | . Saving <None> instead.
2022-01-21T14:24:24.631551528Z prodstack_celeryworker.1.lki36nzyw4k2@analyses    | WARNING 2022-01-21 15:24:24,631 models 123 140487276836224 Traceback: Traceback (most recent call last):
2022-01-21T14:24:24.631598870Z prodstack_celeryworker.1.lki36nzyw4k2@analyses    |   File "/app/topobank/manager/models.py", line 1005, in renew_images
2022-01-21T14:24:24.631606503Z prodstack_celeryworker.1.lki36nzyw4k2@analyses    |     self._renew_images(driver=driver)
2022-01-21T14:24:24.631610738Z prodstack_celeryworker.1.lki36nzyw4k2@analyses    |   File "/app/topobank/manager/models.py", line 955, in _renew_images
2022-01-21T14:24:24.631614835Z prodstack_celeryworker.1.lki36nzyw4k2@analyses    |     image = get_screenshot_as_png(plot, driver=driver)
2022-01-21T14:24:24.631618802Z prodstack_celeryworker.1.lki36nzyw4k2@analyses    |   File "/opt/conda/lib/python3.7/site-packages/bokeh/io/export.py", line 224, in get_screenshot_as_png
2022-01-21T14:24:24.631622808Z prodstack_celeryworker.1.lki36nzyw4k2@analyses    |     web_driver.get("file:///" + tmp.path)
2022-01-21T14:24:24.631626815Z prodstack_celeryworker.1.lki36nzyw4k2@analyses    |   File "/opt/conda/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 333, in get
2022-01-21T14:24:24.631631351Z prodstack_celeryworker.1.lki36nzyw4k2@analyses    |     self.execute(Command.GET, {'url': url})
2022-01-21T14:24:24.631635149Z prodstack_celeryworker.1.lki36nzyw4k2@analyses    |   File "/opt/conda/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 321, in execute
2022-01-21T14:24:24.631639153Z prodstack_celeryworker.1.lki36nzyw4k2@analyses    |     self.error_handler.check_response(response)
2022-01-21T14:24:24.631642996Z prodstack_celeryworker.1.lki36nzyw4k2@analyses    |   File "/opt/conda/lib/python3.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 242, in check_response
2022-01-21T14:24:24.631646960Z prodstack_celeryworker.1.lki36nzyw4k2@analyses    |     raise exception_class(message, screen, stacktrace)
2022-01-21T14:24:24.631650796Z prodstack_celeryworker.1.lki36nzyw4k2@analyses    | selenium.common.exceptions.TimeoutException: Message: Timeout loading page after 300000ms
```
So there is a timeout when internally using geckodriver and loading a page. Maybe this can be increased while running in background?

**Comment by mcrot** [comment]
Probably we need to replace bokeh and selenium at this step. I cannot manage to generate a thumbnail for the 8kx8k topography so far. As a workaround, I'll catch the Runtime exceptions here so there is a proper error message in the log and at least the DZI files will be created.

**Comment by mcrot** [comment]
Even with the workaround roughly 1900 thumbnails cannot be generated for several reasons related to selenium/firefox/geckodriver. We should give #787 a try.

**Comment by mcrot** [comment]
This is fixed by merging PR #775.
